pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -362,17 +197,180 @@ func (b *bootstrapper) WaitCleanup(ctx context.Context) error {
+	var (
+		config string
+		err    error
+	)
+
+	switch b.values.Role {
+	case RoleWeeder:
+		config, err = encodeConfig[weederapi.Config](&b.values.WeederConfig)
+		if err != nil {
+			return nil, err
+		}
+
+	case RoleProber:
+		config, err = encodeConfig[proberapi.Config](&b.values.ProberConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.name() + "-config",
+			Namespace: b.namespace,
+			Labels:    map[string]string{v1beta1constants.LabelApp: b.name()},
+		},
+		Data: map[string]string{configFileName: config},
+	}
+	utilruntime.Must(kubernetesutils.MakeUnique(configMap))
+
+	return configMap, nil
+}
+
+func (b *bootstrapper) getServiceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.name(),
+			Namespace: b.namespace,
+		},
+		AutomountServiceAccountToken: pointer.Bool(false),
+	}
+}
+
+func (b *bootstrapper) getClusterRoleAndClusterRoleBinding(serviceAccNsName types.NamespacedName) (*rbacv1.ClusterRole, *rbacv1.ClusterRoleBinding) {
+
+	clusterRole := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("gardener.cloud:%s:cluster-role", b.name()),
+		},
+		Rules: b.clusterRoleRules(),
+	}
+
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("gardener.cloud:%s:cluster-role-binding", b.name()),
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     clusterRole.Name,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      serviceAccNsName.Name,
+			Namespace: serviceAccNsName.Namespace,
+		}},
+	}
+	return clusterRole, clusterRoleBinding
+}
+
+func (b *bootstrapper) getRoleAndRoleBinding(serviceAccNsName types.NamespacedName) (*rbacv1.Role, *rbacv1.RoleBinding) {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("gardener.cloud:%s:role", b.name()),
+			Namespace: b.namespace,
+		},
+		Rules: b.roleRules(),
+	}
+
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("gardener.cloud:%s:role-binding", b.name()),
+			Namespace: b.namespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     role.Name,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      serviceAccNsName.Name,
+			Namespace: serviceAccNsName.Namespace,
+		}},
+	}
+
+	return role, roleBinding
+}
+
 func (b *bootstrapper) name() string {
-	return fmt.Sprintf("%s-%s", name, b.values.Role)
+	return fmt.Sprintf("%s-%s", prefixDependencyWatchdog, b.values.Role)
 }
 
 func (b *bootstrapper) getLabels() map[string]string {
 	return map[string]string{v1beta1constants.LabelApp: b.name()}
 }
 
+func (b *bootstrapper) roleRules() []rbacv1.PolicyRule {
```

Comment:
Please harmonize the follwing function names.
```suggestion
func (b *bootstrapper) getRolePolicyRules() []rbacv1.PolicyRule {
```